corehq/apps/geospatial/reports.py

```diff
@@ -116,40 +115,148 @@ class CaseGroupingReport(BaseCaseMapReport):
     base_template = 'geospatial/case_grouping_map_base.html'
     report_template_path = 'case_grouping_map.html'
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # geohash grid precision, set by self._aggregate_query(), read
+        # by self.shared_pagination_GET_params()
+        self._precision = None
+
+    def _base_query(self):
+        # Overrides super()._base_query() to not implement pagination
+        # here. It is done later, in self.rows()
+        return self.search_class().domain(self.domain)
+
     def _build_query(self):
-        query = super()._build_query()
-        case_property = get_geo_case_property(self.domain)
+        """
+        Returns a filtered, unaggregated, unpaginated ESQuery.
 
-        # NOTE: ASSUMES polygon is available in request.POST['feature']
-        if 'feature' in self.request.POST:
-            # Filter cases by a shape set by the user
-            geojson = json.loads(self.request.POST['feature'])
+        This allows it to be used by `total_cases()`.
+        """
+        query = super()._build_query()
+        return self._filter_query(query)
+
+    def _filter_query(self, query, additional_filters=None):
+        """
+        Adds ``additional_filters`` to ``query``. If a user-defined
+        polygon (or other GeoShape) is passed in the "feature" GET
+        param, adds that as a filter to ``query``.
+        """
+        # NOTE: Expects GeoShape in request.GET['feature']
+        if self.request.GET.get('feature'):
+            case_property = get_geo_case_property(self.domain)
+            geojson = json.loads(self.request.GET['feature'])
             shape = geojson_to_es_geoshape(geojson)
             relation = 'within' if shape['type'] == 'polygon' else 'intersects'
-            query.nested(
-                CASE_PROPERTIES_PATH,
-                filters.geo_shape(
+            if additional_filters:
+                additional_filters.append(filters.geo_shape(
                     field=case_property,
                     shape=shape,
                     relation=relation,
+                ))
+            else:
+                query.nested(
+                    CASE_PROPERTIES_PATH,
+                    filters.geo_shape(
+                        field=case_property,
+                        shape=shape,
+                        relation=relation,
+                    )
                 )
+        if additional_filters:
+            query.nested(
+                CASE_PROPERTIES_PATH,
+                additional_filters
```

Comment:
I guess this can move in the `if additional_filters` above?

Reply:
The `if additional_filters` above is only executed if the user has selected a polygon. We want `additional_filters` to be applied regardless.

I have refactored this in commit b3523f5b8a3bac0048f42e04297e25504ae59ae9 which I hope makes it clearer.